hackett-doc/scribblings/hackett/reference.scrbl

```diff
@@ -731,6 +897,37 @@ the following expression:
   (foldl * 1 {1 :: 2 :: 3 :: 4 :: 5 :: Nil})
   (foldl - 0 {1 :: 2 :: 3 :: 4 :: 5 :: Nil}))}
 
+@defproc[(unfoldr [step {b t:-> (t:Maybe (t:Tuple a b))}] [seed b]) (t:List a)]{
+
+@racket[unfoldr] constructs a list from an initial value, stopping when @racket[(step seed)] is
+@racket[Nothing]. In a certain way, @racket[unfoldr] acts as a dual to @racket[foldr]. More
+specifically, @racket[{(unfoldr g (foldr f z xs)) == xs}] when @racket[{(g z) == Nothing}] and
+@racket[{(g (f x y)) == (Just (tuple x y))}] for any @racket[x] in @racket[xs].
+
+@(hackett-examples
+  (unfoldr (λ [x] (if {x == 1} Nothing
+                      (Just (Tuple (show x) (quotient! x 2)))))
```

Comment:
If you’re going to use `quotient!` and `remainder!` in examples, they probably need documentation as well. (My policy so far has basically been to document everything undocumented as I use them in other pieces of documentation.)